fog/types/src/view.rs

```diff
@@ -344,16 +354,35 @@ pub struct FogTxOut {
 // Convert a TxOut to a FogTxOut in the efficient way (omitting compressed
 // commitment)
 impl core::convert::From<&TxOut> for FogTxOut {
-    #[inline]
     fn from(src: &TxOut) -> Self {
-        Self {
-            target_key: src.target_key,
-            public_key: src.public_key,
-            amount_masked_value: src.masked_amount.masked_value,
-            amount_masked_token_id: src.masked_amount.masked_token_id.clone(),
-            amount_commitment_data_crc32: Crc::<u32>::new(&crc::CRC_32_ISO_HDLC)
-                .checksum(src.masked_amount.commitment.point.as_bytes()),
-            e_memo: src.e_memo,
+        match src.get_masked_amount() {
+            Err(_) => Self {
```

Comment:
this seems like it could silently introduce bad data. Why not make this `TryFrom`?

Reply:
I started with making this TryFrom, but this created problems in the fog ingest enclave implementation. Basically once we start processing TxOut's, we need the rest of the operation to be no-fail.

We could do all these conversions up-front before processing anything I guess.

I'll revisit this, thanks for the comment